src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -500,7 +509,7 @@ event::ptr primitive_inst::realloc_if_needed() {
     const auto& users = get_user_insts();
     if (users.size() == 1 && users.front()->get_node().is_type<concatenation>()) {
         auto concat_inst = users.front();
-        if (concat_inst->can_be_optimized()) {
+        if (concat_inst->can_be_optimized() && !has_subgraph_dependency(concat_inst->dependencies())) {
```

Comment:
@vladimir-paramuzov , I suspect that current implementation may lead to the accuracy issue, because if concat was marked as optimized_out, probably it expects the proper output buffer filling by its' input primitives, but not setting usm_device memory to add operation will prevent this
So, I suggested to Michal that concat optimization should be disabled for cases where concat has input nodes from ShapeOf-subgraphs

Reply:
I disabled auto-merge for now